pkg/virt-handler/isolation/isolation.go

```diff
@@ -215,34 +216,53 @@ func getMemlockSize(vm *v1.VirtualMachineInstance) (int64, error) {
 	return bytes_, nil
 }
 
-func NewIsolationResult(pid int, slice string, controller []string) *IsolationResult {
-	return &IsolationResult{pid: pid, slice: slice, controller: controller}
+func NewIsolationResult(pid int, slice string, controller []string) IsolationResult {
+	return &RealIsolationResult{pid: pid, slice: slice, controller: controller}
 }
 
-type IsolationResult struct {
+type IsolationResult interface {
+	Slice() string
+	Pid() int
+	PIDNamespace() string
+	MountRoot() string
+	MountInfoRoot() (*MountInfo, error)
+	MountNamespace() string
+	NetNamespace() string
+	DoNetNS(func(_ ns.NetNS) error) error
+}
+
+type RealIsolationResult struct {
 	pid        int
 	slice      string
 	controller []string
 }
 
-func (r *IsolationResult) Slice() string {
-	return r.slice
+func (r *RealIsolationResult) DoNetNS(f func(_ ns.NetNS) error) error {
+	netns, err := ns.GetNS(r.NetNamespace())
```

Comment:
@rmohr i know we had a lot of discussions about the general safety of attempting to switch between namespaces without forking a process. I haven't kept up with developments in that conversation. Can you verify this is safe

Reply:
I did not yet look into the implementation here. In general it is now save for as long as you do runtime.LockOSThread() and don't spawn go routines in the locked section: https://www.weave.works/blog/linux-namespaces-golang-followup